Lib/multiprocessing/util.py

```diff
@@ -135,7 +150,26 @@ def get_temp_dir():
     tempdir = process.current_process()._config.get('tempdir')
     if tempdir is None:
         import shutil, tempfile
-        tempdir = tempfile.mkdtemp(prefix='pymp-')
+        if os.name == 'nt':
+            tempdir = tempfile.mkdtemp(prefix='pymp-')
+        else:
+            # Most of the time, the root temporary directory is /tmp, and thus
+            # listener sockets files "$TMPDIR/pymp-XXXXXXXX/sock-XXXXXXXX"
+            # do not have a path length exceeding SUN_PATH_MAX.
+            #
+            # If users specify their own temporary directory, we may be unable
+            # to create those files. Therefore, we fall back to the system-wide
+            # temporary directory /tmp, assumed to exist on POSIX systems.
+            #
+            # See https://github.com/python/cpython/issues/132124.
+            base_tempdir = tempfile.gettempdir()
+            # len(base_tempdir) + len('/pymp-XXXXXXXX') + len('/sock-XXXXXXXX')
+            sun_path_len = len(base_tempdir) + 14 + 14
+            if sun_path_len > _SUN_PATH_MAX:
+                # fallback to the system-wide temporary directory,
+                # ignoring environment variables.
+                base_tempdir = '/tmp'
```

Comment:
Before doing this, perhaps test that /tmp exists and we have permissions to create files within it.

that'd be a situation where using the warnings module to emit a warning about being unable to use /tmp and base_tempdir potentially being too long to create a unix socket.

Reply:
> perhaps test that /tmp exists and we have permissions to create files within it.

Good catch. Though, that's why I actually talked about POSIX systems, but it doesn't hurt checking at runtime to prevent obscure errors.